src/main/scala/sbtdatabricks/DatabricksPlugin.scala

```diff
@@ -183,8 +187,70 @@ object DatabricksPlugin extends AutoPlugin {
     }
   }
 
+  private lazy val executeCommandImpl: Def.Initialize[Task[Seq[CommandStatus]]] = Def.task {
+    val client = dbcApiClient.value
+    val language = dbcExecutionLanguage.value
+    val onClusters = dbcClusters.value
+    val (allClusters, _) = dbcFetchClusters.value
+    val commandFile = dbcCommandFile.value
+    val commandStatuses = Seq.empty[CommandStatus]
+
+    @annotation.tailrec
+    def onContextCompletion(contextId: ContextId, cluster: Cluster) : Option[ContextId] = {
+      val contextStatus = client.checkContext(contextId, cluster)
+      contextStatus.status match {
+        case DBCContextRunning.status => Some(contextId)
+        case DBCContextError.status => {
+          client.destroyContext(contextId, cluster)
+          None
+        }
+        case _ => {
+          Thread sleep 500
+          onContextCompletion(contextId, cluster)
+        }
+      }
+    }
+
+    @annotation.tailrec
+    def onCommandCompletion(cluster: Cluster, contextId: ContextId, commandId: CommandId) : Option[CommandId] = {
+      val commandStatus = client.checkCommand(cluster, contextId, commandId)
+      commandStatus.status match {
+        case DBCCommandFinished.status => {
+          commandStatuses :+ commandStatus
+          Some(commandId)
+        }
+        case DBCCommandError.status => {
+          client.cancelCommand(cluster, contextId, commandId)
+          client.destroyContext(contextId, cluster)
+          None
+        }
+        case _ => {
+          Thread sleep 3000
+          onCommandCompletion(cluster, contextId, commandId)
+        }
+      }
+    }
+
+    client.foreachCluster(onClusters, allClusters) { confirmedCluster =>
+      val contextId = onContextCompletion(
+        client.createContext(language, confirmedCluster),
+        confirmedCluster)
+
+      if (contextId != None) {
```

Comment:
A more scala way of doing this could be:
```scala
contextId.foreach { cId =>
  val commandId = onCommandCompletion(
    confirmedCluster,
    cId,
    client.executeCommand(language, confirmedCluster, cId, commandFile))
    if (commandId.isDefined) {
      client.destroyContext(cId, confirmedCluster)
    }
}
```